packages/frontend-2/components/project/page/settings/general/General.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div class="flex flex-col gap-4">
+    <ProjectPageSettingsGeneralBlockProjectInfo
+      v-if="project"
+      :project="project"
+      :disabled="isDisabled"
+      @update-project="
+        ({ name, description }) =>
+          handleUpdate({ name, description }, 'Project info updated')
+      "
+    />
+    <ProjectPageSettingsGeneralBlockAccess
+      v-if="project"
+      :project="project"
+      :disabled="isDisabled"
+      @update-visibility="
+        (newVisibility) =>
+          handleUpdate({ visibility: newVisibility }, 'Project access updated')
+      "
+    />
+    <ProjectPageSettingsGeneralBlockDiscussions
+      v-if="project"
+      :project="project"
+      :disabled="isDisabled"
+      @update-comments-permission="
+        (newCommentsPermission) =>
+          handleUpdate(
+            { allowPublicComments: newCommentsPermission },
+            'Comment permissions updated'
+          )
+      "
+    />
+    <ProjectPageSettingsGeneralBlockLeave
+      v-if="canLeaveProject && project"
+      :project="project"
+    />
+
+    <ProjectPageSettingsGeneralBlockDelete
+      v-if="project && isOwner && !isGuest"
+      :project="project"
+      @update-comments-permission="
+        (newCommentsPermission) =>
+          handleUpdate(
+            { allowPublicComments: newCommentsPermission },
+            'Comment permissions updated'
+          )
+      "
+    />
+  </div>
+</template>
+<script setup lang="ts">
+import { Roles } from '@speckle/shared'
+import { useQuery } from '@vue/apollo-composable'
+import type { ProjectUpdateInput } from '~~/lib/common/generated/gql/graphql'
+import { useUpdateProject } from '~~/lib/projects/composables/projectManagement'
+import { graphql } from '~~/lib/common/generated/gql'
+
+const projectPageSettingsGeneralQuery = graphql(`
+  query ProjectPageSettingsGeneral($projectId: String!) {
+    project(id: $projectId) {
+      id
+      team {
```

Comment:
This bit only seems to be relevant in the Leave block, so it should be moved into its fragment and then "canLeaveProject" should be a computed inside of there